transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageController.groovy

```diff
@@ -0,0 +1,66 @@
+package org.transmartproject.rest
+
+import grails.rest.RestfulController
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.core.exceptions.AccessDeniedException
+import org.transmartproject.core.ontology.StudiesResource
+import org.transmartproject.core.users.UsersResource
+import org.transmartproject.db.storage.LinkedFileCollection
+import org.transmartproject.db.user.User
+import org.transmartproject.rest.misc.CurrentUser
+
+import javax.annotation.Resource
+
+import static org.transmartproject.core.users.ProtectedOperation.WellKnownOperations.READ
+
+/**
+ * Created by piotrzakrzewski on 02/12/2016.
+ */
+@Slf4j
+class StorageController extends RestfulController {
+
+    static responseFormats = ['json']
+
+    @Autowired
+    CurrentUser currentUser
+
+    @Resource
+    StudiesResource studiesResourceService
+
+    @Autowired
+    UsersResource usersResource
+
+    StorageController() {
+        super(LinkedFileCollection)
+    }
+
+    @Override
+    def show() {
+        def fileCollection = queryForResource(params.id)
```

Comment:
Or add is as parameter to the function (`def show(Long id)` or something like that). But I'm never sure which method is best in Grails, so choose your own preferred way.